### ISSUE TITLE
RET-6 — Delete the `mace/tools` god-module, the last of legacy `mace/`, and the launcher's legacy branch

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1602]
**Depends on:** RET-1 (#1597), RET-2 (#1598), RET-3 (#1599), RET-4 (#1600), RET-5 (#1601), REL-1 (#1603), INF-5 (#1554) · **Blocks:** REL-2 (#1604), REL-4 (#1606)

**Context:** This is the **terminal deletion-only PR** of the strangler migration. Throughout the rewrite, a byte-frozen legacy `mace/` package — **83** tracked files at `3b43d63f` — lived alongside `packages/{mace-core,mace-torch,mace-jax}` as the numerical oracle, and a ~50-LOC launcher (`mace_launcher`, INF-2 (#1551)) owned every console entry point and dispatched `--engine {legacy,v1}` / `MACE_ENGINE` to `mace.cli.*` (legacy) or `mace_torch.cli.*` (v1), defaulting to `legacy` on day one. Capabilities were retired one axis at a time — energy (RET-1 (#1597)) → data layer (RET-2 (#1598)) → the five `convert_*` CLIs (RET-3 (#1599)) → dipole/polar/magnetic/`extensions.py` (RET-4 (#1600)) → LAMMPS jit + legacy calculators (RET-5 (#1601)). Each of those deleted a subsystem after it had been default-`v1` through a deprecation window with a green parity job, and degraded its parity test from "in-process vs live legacy" to "vs frozen golden".

After RET-1 (#1597)..RET-5 (#1601), what is left under `mace/` is the **`mace/tools` god-module** — the training infrastructure that was never a v1 dependency: `arg_parser.py` (**1417 lines**), `train.py` (including the `--lbfgs` regime's `take_step_lbfgs` at `:464`), `scripts_utils.py`, `checkpoint.py`, `compile.py`, `multihead_tools.py`, `finetuning_utils.py`, `run_train_utils.py`, `model_script_utils.py`, `arg_parser_tools.py`, `tables_utils.py`, `torch_tools.py`, `distributed_tools.py`, `slurm_distributed.py`, `default_keys.py`, `cg.py`, `scatter.py`, the vendored `fairchem_dataset/`, and whatever remains of `torch_geometric/` — plus the surviving `mace/cli/*` (`run_train.py`, `eval_configs.py`, `preprocess_data.py`, `select_head.py`, `plot_train.py`, `visualise_train.py`, `active_learning_md.py`, `fine_tuning_select.py`, `convert_device.py`, `polar_density_cube.py`), the remaining `mace/calculators/*` (`mace.py`, `mace_torchsim.py`, `foundations_models.py` and the four bundled artifacts under `foundations_models/`), the residual `mace/modules/*`, and the package glue `mace/__init__.py`, `mace/__version__.py`, `mace/py.typed`. This ticket deletes **all of it** and rips the now-dead legacy branch out of the launcher, driving `git ls-files mace/` to **0** — the retirement criterion the whole plan is gated on. It carries **zero new v1 logic**: it only removes files, flips two `capabilities.toml` rows, and simplifies the launcher/import-guard now that there is no legacy zone left to fence off.

**Interface & constraints:**

1. **Deletion-only budget (binding).** The diff is pure deletion + CI/manifest bookkeeping. No behaviour of the v1 stack changes; no file under `packages/**` gains logic. The only non-deletion edits allowed are: (a) `capabilities.toml` state flips, (b) removal of the launcher's legacy dispatch branch, (c) collapsing the import-guard/toolchain scaffolding that only existed to police the boundary, (d) renaming the `mace-torch-v1` distribution to `mace-torch` — this is the PR that frees the name, so it is also the only place the rename can happen before REL-2 (#1604) publishes.

2. **The terminal invariant.** `git ls-files mace/ == 0` after this PR, and CI enforces it as a hard, permanent gate (it was "strictly non-increasing" from Phase 3 onward; here it reaches zero from a baseline of 83). This is exactly the count the REL-2 (#1604) 1.0-tag release gate reads — REL-2 (#1604) stays blocked until it is 0, so this ticket is what unblocks the tag.

3. **The four bundled model artifacts go with the tree, and that is a size event, not a detail.** `mace/calculators/foundations_models/` ships `2023-12-03-mace-mp.model`, `ani500k_large_CC.model`, `mace-mpa-0-medium.model` and `mp_vasp_e0.json` inside the distribution — `mace_anicc` (`mace/calculators/foundations_models.py:454-491`) loads its checkpoint straight out of the package. v1 downloads foundation artifacts through the FM-3 (#1588) registry instead of vendoring them, and MACE-ANI-CC is a recorded DROP (REL-1 (#1603)). Confirm the FM-3 (#1588) registry serves every artifact a user could previously get from the installed package before deleting them, and record the sdist/wheel size change on the PR.

4. **Launcher legacy branch removed.** `mace_launcher` stops routing to `mace.cli.*`. `--engine=v1` becomes the sole engine: either drop the `--engine`/`MACE_ENGINE` selector entirely, or keep it and make `legacy` a **hard, loud error** that points users at the last v0.3.x release — never a silent fallback to a deleted module. `mace_launcher` remains the single owner of the `mace_*` console scripts (that ownership decision does not change).

5. **Import-guard collapses to trivially-satisfied (INF-3 (#1552)).** The `import-linter` *forbidden* contract `packages → mace` is now vacuously true (there is no `mace` to import); the runtime `sys.addaudithook` reach-in check for `packages.* → mace.*` is removed; the dual-import allowlist (`mace_launcher`, `tests/parity/`) no longer needs the legacy exception. Keep the *layers* contracts (`mace_core ⊥ torch/jax/e3nn`, `mace_jax ⊥ mace_torch`) — those outlive legacy.

6. **Single toolchain (INF-4 (#1553)).** The path-scoped dual toolchain existed only because `mace/**` used black+isort+pylint+mypy (`setup.cfg`'s `[dev]` extra pins `pylint==4.0.6`; `pyproject.toml` carries the pylint/black/isort config and `.mypy.ini` the mypy config) while `packages/**` uses ruff+ty. With `mace/**` gone, retire the `lint` job's legacy half in `ci-core.yaml`, the vendored-code coverage `omit` list (`pyproject.toml:37-43`, which exists only to exclude `mace/tools/torch_geometric/*` and `mace/tools/scatter.py`), the black/isort/pylint/mypy configuration blocks, and the 1:1-ownership meta-lint's legacy half; one toolchain (ruff + ty + prek) governs the whole tree.

7. **Parity already degraded — nothing to degrade here.** Unlike RET-1 (#1597)..RET-5 (#1601), this PR does not degrade a live-oracle parity test to a frozen golden: RET-1 (#1597)..RET-5 (#1601) already did that for every migrated capability. `mace/tools` was training scaffolding, not a numeric oracle capability, so there is no in-process legacy comparison left to sever. Just confirm no `tests/parity/` test still imports a live legacy module.

8. **Preconditions (from scope).** All of RET-1 (#1597)..RET-5 (#1601) merged, **plus the REL gate**: the migration guide (REL-1 (#1603)) must be published, so users have a documented path before legacy leaves the tree. There is no maintenance branch: v0.3 lives on `main` until v1 ships, and after that its home is its final release, pinned from PyPI. Legacy's home is therefore a released version, not a branch.

**Task:**
1. Confirm RET-1 (#1597)..RET-5 (#1601) are merged and that `git ls-files mace/` now contains only `mace/tools/**`, the surviving `mace/cli/**` and `mace/calculators/**`, residual `mace/modules/**` and the three top-level glue files; confirm the migration guide (REL-1 (#1603)) is published and the FM-3 (#1588) registry serves every previously-bundled artifact.
2. Delete the entire `mace/` tree — `mace/tools/` (including the vendored `torch_geometric/`, `scatter.py` and `fairchem_dataset/`), the remaining `mace/cli/*` (including `convert_device.py`, `polar_density_cube.py`, `visualise_train.py`), `mace/calculators/*` and its four bundled artifacts, residual `mace/modules/*`, and `mace/__init__.py` / `mace/__version__.py` / `mace/py.typed` — so `git ls-files mace/ == 0`.
3. Remove the launcher's legacy dispatch branch in `mace_launcher`: v1 is the only engine; `--engine=legacy` / `MACE_ENGINE=legacy` either disappears or hard-errors pointing at the last v0.3.x release. Keep `mace_launcher` as sole owner of the console scripts.
4. `capabilities.toml` (INF-5 (#1554)): flip the god-module and launcher-legacy-branch capability rows to `retired`; remove the matching `xfail(strict=True)` fitness function; regenerate CI path-filters and per-package coverage floors so nothing references `mace/**` (coverage floors now over `packages/` only, and `[tool.coverage.run] source = ["mace"]` no longer points at the deleted tree).
5. INF-3 (#1552) guard cleanup: drop the `forbidden(packages→mace)` machinery's legacy target and the audit-hook reach-in check (or leave both as trivially-green no-ops); collapse the two-module dual-import allowlist. Keep the layers contracts.
6. INF-4 (#1553) toolchain cleanup: retire the legacy lint over `mace/**`, the black/isort/pylint/mypy config blocks, `.mypy.ini`, the vendored-code coverage `omit` list, and the legacy half of the ownership meta-lint; single toolchain over `packages/**`.
7. Confirm every `tests/parity/` test runs against frozen goldens with no live-legacy import; run the full `packages/` suite.

**Out of scope:** the 1.0 tag and its hard release gate (REL-2 (#1604)); the `packages/ → mace/` container rename (REL-4 (#1606), after the 1.0 publish, once this frees the `mace` name); the migration guide (REL-1 (#1603), a precondition, not this PR); any new v1 logic whatsoever.

**Acceptance criteria:**
- [ ] `git ls-files mace/` returns **0** files (down from 83); the CI gate enforcing it is permanent and green.
- [ ] The diff is deletion-only: file removals + `capabilities.toml` flips + launcher-branch removal + guard/toolchain scaffolding collapse — **zero new v1 logic**.
- [ ] `--engine=legacy` / `MACE_ENGINE=legacy` no longer route anywhere; v1 is the sole engine (the flag is gone, or `legacy` is a loud error pointing at the last v0.3.x release, never a silent fallback).
- [ ] Every artifact previously bundled under `mace/calculators/foundations_models/` is reachable through the FM-3 (#1588) registry, or is a recorded DROP; the distribution size change is noted on the PR.
- [ ] `import-linter` is green with the `packages → mace` forbidden contract now trivially satisfied; the runtime audit-hook's `packages→mace` reach-in check is removed; the dual-import allowlist no longer carries a legacy exception.
- [ ] `capabilities.toml`: the god-module and launcher-legacy rows read `retired`; the matching `xfail(strict=True)` fitness function is removed; a **single** toolchain governs the tree (no legacy lint step, no black/isort/pylint/mypy config, no `.mypy.ini`); coverage floors and `[tool.coverage.run] source` are over `packages/` only.
- [ ] No `tests/parity/` (or any) test imports a live legacy `mace` module; all parity tests run vs frozen goldens.
- [ ] The `packages/` suite and lint are green; the REL-2 (#1604) release gate can now read `git ls-files mace/ == 0`.

**Verify:**
```bash
test "$(git ls-files mace/ | wc -l | tr -d ' ')" -eq 0        # the terminal invariant (was 83)
lint-imports --config .importlinter                           # forbidden(packages→mace) now trivially satisfied
python -m pytest tests/parity                                 # all vs frozen goldens; no live-legacy import
python -m pytest packages/*/tests
mace_run_train --engine legacy 2>&1 | grep -qi "v0.3" # legacy engine removed → points at the released v0.3 line
grep -rn "import mace\b\|from mace " packages/ tests/ ; test $? -ne 0   # no lingering legacy import
```
**Review focus:** that the diff is strictly deletion + bookkeeping (no smuggled v1 logic); that `git ls-files mace/ == 0` is wired as a hard CI gate the REL-2 (#1604) tag reads; that no entry point, import path, audit-hook, or test still reaches into a now-deleted legacy module; that the four bundled foundation artifacts have a registry home or a recorded DROP before they vanish; and that the launcher fails loudly on `--engine=legacy` rather than silently.

---
